e2e/lib/testing.sh

```diff
@@ -23,10 +23,20 @@ function testing_run_test {
     local testname=$(testing_get_test_metadata "$testfile" "TEST-NAME")
     int_start=$(date +%s)
     echo "+++++ $(date): starting $testfile $testname"
-    /bin/bash "$testfile"
-    echo "+++++ $(date): finished $testfile $testname"
-    printf "TIME $(basename $testfile): %s secs\n" "$(($(date +%s) - int_start))"
+    local rc=0
+    /bin/bash "$testfile" || rc=$?
+    local result="PASS"
+    if [[ $rc != 0 ]]; then
+        result="FAIL ($rc)"
+    fi
 
-    echo "Porch Controller logs"
-    kubectl logs deployment/porch-controllers -n porch-system --since "$(($(date +%s) - int_start))s" | sed -e '/PackageVariant/!d;/resources changed/!d'
+    echo "+++++ $(date): finished $testfile $testname (result: $result)"
+    local seconds="$(($(date +%s) - int_start))"
+    printf "TIME $(basename $testfile): %s secs\n" $seconds
+    test_summary="${test_summary-}$(echo && echo $(basename $testfile): $result in $seconds seconds)"
```

Comment:
You can also concatenate values and print them with `echo -e $test_summary`
```suggestion
    test_summary+="$(basename $testfile): $result in $seconds seconds\n"
```

Reply:
"\n" doesn't work in bash, what's the better alternative? I have this:
```
[hi on] jbelamaric@jbelamaric:~/proj/gh/johnbelamaric/nephio-test-infra/e2e$ unset test
[hi on] jbelamaric@jbelamaric:~/proj/gh/johnbelamaric/nephio-test-infra/e2e$ test+="$(basename /tmp/foo): testing\n" && echo "$test"
foo: testing\n
[hi on] jbelamaric@jbelamaric:~/proj/gh/johnbelamaric/nephio-test-infra/e2e$ test+="$(basename /tmp/foo): testing\n" && echo "$test"
foo: testing\nfoo: testing\n
[hi on] jbelamaric@jbelamaric:~/proj/gh/johnbelamaric/nephio-test-infra/e2e$ test+="$(echo && echo $(basename /tmp/foo): testing)" && echo "$test"
foo: testing\nfoo: testing\n
foo: testing
[hi on] jbelamaric@jbelamaric:~/proj/gh/johnbelamaric/nephio-test-infra/e2e$ test+="$(echo && echo $(basename /tmp/foo): testing)" && echo "$test"
foo: testing\nfoo: testing\n
foo: testing
foo: testing
[hi on] jbelamaric@jbelamaric:~/proj/gh/johnbelamaric/nephio-test-infra/e2e$ test+="$(echo && echo $(basename /tmp/foo): testing)" && echo "$test"
foo: testing\nfoo: testing\n
foo: testing
foo: testing
foo: testing
[hi on] jbelamaric@jbelamaric:~/proj/gh/johnbelamaric/nephio-test-infra/e2e$ 
```
but I agree that's awkward. I swear in /sh or maybe it was csh or ksh or something, \n used to work years ago...

Reply:
It works if you use `-e` flag in the `echo` instruction. 

```bash
$ unset test
$ test+="$(basename /tmp/foo): testing\n" && echo -e "$test"
foo: testing

$ test+="$(basename /tmp/foo): testing\n" && echo -e "$test"
foo: testing
foo: testing

```